src/interactions/slashCommands/mixins/CommandOptionWithChoices.ts

```diff
@@ -1,16 +1,12 @@
 import { APIApplicationCommandOptionChoice, ApplicationCommandOptionType } from 'discord-api-types/v9';
-import ow, { Predicate } from 'ow';
+import { z } from 'zod';
 import { validateMaxChoicesLength } from '../Assertions';
 import type { ToAPIApplicationCommandOptions } from '../SlashCommandBuilder';
 import { SlashCommandOptionBase } from './CommandOptionBase';
 
-const stringPredicate = ow.string.minLength(1).maxLength(100);
-const integerPredicate = ow.number.finite;
-
-// TODO: See resolution for sindresorhus/ow#217 in relation to this cast
-const choicesPredicate = ow.array.ofType<[string, string | number]>(
-	ow.array.exactShape([stringPredicate, ow.any(ow.string, integerPredicate) as unknown as Predicate<string | number>]),
-);
+const stringPredicate = z.string().min(1).max(100);
+const integerPredicate = z.number().gt(-Infinity).lt(Infinity);
```

Comment:
From https://github.com/colinhacks/zod#numbers:
```suggestion
const integerPredicate = z.number().int();
```

Reply:
The predicate needs to allow floats for number options.

Reply:
Then it's not an integer predicate, please rename the constant.